src/components/OrientationAwareIcon.tsx

```diff
@@ -17,26 +18,45 @@ export const OrientationAwareIcon = (props: Props) => {
   const { orientation, dimensions } = useContext(OrientationContext);
   const {
     Icon,
+    iconName,
     iconStyle,
     landscapeIconStyle,
     landscapeStyle,
     size = normalize(24),
     style
   } = props;
 
+  const SelectedIcon = Icon || IconComponent.NamedIcon;
+
   const needLandscapeStyle =
     orientation === 'landscape' || dimensions.width > consts.DIMENSIONS.FULL_SCREEN_MAX_WIDTH;
 
   // we need adjustments only on iOS, so otherwise just return the item with the usual props
   // TODO: is this still needed with Expo 48?
   if (!needLandscapeStyle || !(device.platform === 'ios')) {
-    return <Icon {...props} />;
+    if (SelectedIcon === IconComponent.NamedIcon) {
+      return <SelectedIcon {...props} name={iconName} />;
+    }
+
+    return <SelectedIcon {...props} />;
```

Comment:
can you think of some refactoring to have this not that duplicated and with an own if?

Reply:
I greatly reduced code repetition.

[fa2b69f](https://github.com/smart-village-solutions/smart-village-app-app/pull/1226/commits/fa2b69f36d20317638b6c5f16cdd1891288aaa54)